src/BookNavigator/bookmarks/ia-bookmarks.js

```diff
@@ -444,27 +450,34 @@ class IABookmarks extends LitElement {
       <button
         class="ia-button primary"
         ?disabled=${enableAddBookmark}
-        @click=${this.addBookmark}
-      >Add bookmark</button>
+        @click=${this.addBookmark}>
+        Add bookmark
+      </button>
     `;
 
+    const bookmarkSignedInLayout = html`
+      <ia-bookmarks-list
+        @bookmarkEdited=${this.bookmarkEdited}
+        @bookmarkSelected=${this.bookmarkSelected}
+        @saveBookmark=${this.saveBookmark}
+        @deleteBookmark=${this.deleteBookmark}
+        .editedBookmark=${this.editedBookmark}
+        .bookmarks=${{ ...this.bookmarks }}
+        .activeBookmarkID=${this.activeBookmarkID}
+        .bookmarkColors=${this.bookmarkColors}
+        .defaultBookmarkColor=${this.defaultColor}>
+      </ia-bookmarks-list>
+      ${this.allowAddingBookmark ? addBookmarkButton : nothing}
+    `
+
+    const bookmarkLoginLayout = html`
+      <p>A free account is required to save and access bookmarks.</p>
+      <button class="ia-button primary" @click=${this.bookmarkOptions.handleLogin}>Login</button>
+    `
     return html`
       <section class="bookmarks">
-        <ia-bookmarks-list
-          @bookmarkEdited=${this.bookmarkEdited}
-          @bookmarkSelected=${this.bookmarkSelected}
-          @saveBookmark=${this.saveBookmark}
-          @deleteBookmark=${this.deleteBookmark}
-
-          .editedBookmark=${this.editedBookmark}
-          .bookmarks=${{ ...this.bookmarks }}
-          .activeBookmarkID=${this.activeBookmarkID}
-
-          .bookmarkColors=${this.bookmarkColors}
-          .defaultBookmarkColor=${this.defaultColor}
-        ></ia-bookmarks-list>
-        ${this.allowAddingBookmark ? addBookmarkButton : nothing}
-      </section
+        ${this.displayMode === 'login' ? bookmarkLoginLayout : bookmarkSignedInLayout }        
```

Comment:
Can we say these const like `bookmarkLoginLayout` and `bookmarkLoggedInLayout`. login and signedIn seems not correct?

Reply:
refactored to be more semantic and readable